node/agent.go

```diff
@@ -211,13 +211,36 @@ func attachAndRun(dockerID, command string) error {
 	} else if command == "" {
 		return nil
 	}
+	type out struct {
+		response string
+		err      error
+	}
+	done := make(chan out)
 
-	output, err := utils.AttachAndRun(dockerID, []string{command})
-	if err != nil {
-		err = fmt.Errorf("%s (%s)", string(output), err)
-		glog.Errorf("Could not pause container %s: %s", dockerID, err)
+	go func() {
+		output, err := utils.AttachAndRun(dockerID, []string{command})
+		done <- out{string(output), err}
+	}()
+
+	select {
+		case result := <- done: {
+			if result.err != nil {
+				glog.Errorf("Could not pause container %s: %s", dockerID, result.err)
+				return result.err
+			}
+		}
+		// replace hardcoded timeout with f.dfs.Timeout() - 10 sec
+		case <- time.After(time.Duration(290 * time.Second)): {
+			kill := "quiesce=$(ps aux | grep '.*quiesce.*sh.*' | grep -v grep | awk '{print $2}');" +
```

Comment:
Since this is the generic attachAndRun that could be used for anything, you should use the passed-in command to build the kill command.